lib/nifty_services/base_create_service.rb

```diff
@@ -24,86 +23,75 @@ def execute
     end
 
     private
+
     def save_record
       begin
         @record.save
       rescue => e
         on_save_record_error(e)
-        return false
+      ensure
+        @record.valid?
       end
     end
 
     def on_save_record_error(error)
-      return unprocessable_entity_error!(error)
+      unprocessable_entity_error!(error)
     end
 
     def create_error_response(record)
       record.errors
     end
 
     def after_error_response(errors)
-      unprocessable_entity_error(errors) if errors.present?
+      unprocessable_entity_error(errors) unless errors.empty?
     end
 
     def after_execute_success_response
       success_created_response
     end
 
     def build_record
-      if record_type.present?
+      unless record_type.nil?
         return build_from_record_type(record_allowed_attributes)
       end
 
-      return not_implemented_exception(__method__)
+      not_implemented_exception(__method__)
+    end
+
+    def build_record_scope
+      nil
     end
 
     def build_from_record_type(params)
       if !build_record_scope.nil? && build_record_scope.respond_to?(:build)
-        return build_record_scope.send(:build, params)
+        return build_record_scope.build(params)
       end
 
-      record_type.send(:new, params)
+      record_type.new(params)
     end
 
     def can_execute?
-      if validate_ip_on_create? && !can_create_with_ip?
-        return forbidden_error!(%s(users.ip_temporarily_blocked))
-      end
-
       if validate_user? && !valid_user?
         return not_found_error!(invalid_user_error_key)
       end
 
-      return true
+      true
     end
 
     def can_create_record?
       unless user_can_create_record?
         return (valid? ? forbidden_error!(user_cant_create_error_key) : false)
       end
 
-      return true
+      true
```

Comment:
@viniciuszago Same here, above the code to some flow control using return, so explicity return here makes things more clear.